### PR TITLE
Add annotations

### PR DESCRIPTION
**Pull request by soc**

### CODE REVIEW

**Comment by soc** [comment]
@dinfuehr could you have a look?

File: dora-parser/src/ast/dump.rs [hunk 1]
```diff
@@ -297,6 +298,42 @@ impl<'a> AstDumper<'a> {
...
+        });
+    }
+
+    fn dump_annotation_param(&mut self, param: &AnnotationParam) {
+        dump!(self, "param {} @ {}", self.str(param.name), param.pos);
+
+        self.indent(|d| d.dump_type(&param.data_type));
+    }
+
+    fn dump_annotation_usages(&self, annotation_usages: &AnnotationUsages) {
```
Comment (dinfuehr) [inline dora-parser/src/ast/dump.rs:326]: please fix this warning

File: dora-parser/src/ast.rs [hunk 2]
```diff
@@ -602,9 +637,11 @@ impl Function {
     }
 }
 
+//remove in next step
```
Comment (dinfuehr) [inline dora-parser/src/ast.rs:640]: IMHO it's not necessary to add these comments, but if you do please add a space after the slashes

File: dora-parser/src/ast/dump.rs (hunk 1 above)
Comment (soc) [reply dora-parser/src/ast/dump.rs:327]: Added a `#[allow(dead_code)]` until it is used, not sure why it isn't showing up here.

**Comment by soc** [comment]
@dinfuehr Could you have a look when it turns green?

**Comment by dinfuehr** [comment]
did you add more commits now?

**Comment by soc** [comment]
@dinfuehr Yes, I finished splitting the patch into smaller parts, everything in the big patch should now be in this PR.

**Comment by soc** [comment]
@dinfuehr rebased!

**Comment by dinfuehr** [comment]
Please don't add more commits to the PR after it was already reviewed. Let's revert to the commits I looked at and land those first. Then try to land the rest in one or more subequent PRs.

**Comment by soc** [comment]
@dinfuehr Removed commits, fixed conflicts, rebased.

**Comment by dinfuehr** [comment]
Thanks!

**Pull request merged**